File: frontend/src/model/entities/Analise.ts
import type { RespostaAvaliacao } from "./RespostaAvaliacao";

export interface Analise {
  id?: number;

  eixoMaxAtitude?: number;
  eixoMaxCapacidade?: number;

  liderancaComandanteTotal?: number;
  liderancaTreinadorTotal?: number;
  liderancaOrientadorTotal?: number;
  liderancaDesafiadorTotal?: number;

  eixoAtitude?: number;
  eixoCapacidade?: number;

  eixoLiderancaComandante?: number;
  eixoLiderancaTreinador?: number;
  eixoLiderancaOrientador?: number;
  eixoLiderancaDesafiador?: number;

  resumoIA?: string;
  respostaAvaliacao?: RespostaAvaliacao;
}

File: backend/src/main/java/com/hr_solution/model/entities/Analise.java
package com.hr_solution.model.entities;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "analise")
public class Analise {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //Eixos do Perfil

    //Máximo atingível no eixo da Atitude
    @Column(nullable = true)
    private Integer EixoMaxAtitude;

    @Column(nullable = true)
    private Integer EixoMaxCapacidade;

    //Eixos da Liderança

    //Máximo atingível no eixo da Liderança Comandante
    @Column(nullable = true)
    private Integer LiderancaComandanteTotal;

    @Column(nullable = true)
    private Integer LiderancaTreinadorTotal;
    
    @Column(nullable = true)
    private Integer LiderancaOrientadorTotal;

    @Column(nullable = true)
    private Integer LiderancaDesafiadorTotal;

    //Eixos do Perfil

    //Valor atingido nos eixos de Perfil
    @Column(nullable = true)
    private Integer eixoAtitude;

    @Column(nullable = true)
    private Integer eixoCapacidade;

    //Eixos da Liderança

    //Valor atingido nos eixos de Liderança
    @Column(nullable = true)
    private Integer eixoLiderancaComandante;

    @Column(nullable = true)
    private Integer eixoLiderancaTreinador;
    
    @Column(nullable = true)
    private Integer eixoLiderancaOrientador;

    @Column(nullable = true)
    private Integer eixoLiderancaDesafiador;

    @Column(nullable = true)
    private String resumoIA;

    @OneToOne
    @JoinColumn(name = "resposta_avaliacao_id")
    private RespostaAvaliacao respostaAvaliacao;
}
